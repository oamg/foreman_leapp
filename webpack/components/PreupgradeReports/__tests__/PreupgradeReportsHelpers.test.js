import { testSelectorsSnapshotWithFixtures } from '@theforeman/test';

import {
  preupgradeReports,
  reportsWithRemediations,
} from './PreupgradeReports.fixtures';

import {
  flattenEntries,
  isEmpty,
  entryFixable,
  anyEntriesFixable,
  idsForInvocationFromReports,
  entriesPage,
  filterEntries,
  fixableEntries,
  sortEntries,
} from '../PreupgradeReportsHelpers';

const fixtures = {
  'should return flattened entrues': () => flattenEntries(preupgradeReports),
  'should detect empty object': () => isEmpty({}),
  'should detect non-empty object': () => isEmpty({ foo: 'bar' }),
  'should detect entry is fixable': () =>
    entryFixable({ detail: { remediations: [{ type: 'command' }] } }),
  'should detect entry is not fixable when not having a command': () =>
    entryFixable({ detail: { remediations: [{ type: 'hint' }] } }),
  'should detect entry is not fixable when not having remediations': () =>
    !!entryFixable({ detail: {} }),
  'should detect no entry is fixable': () =>
    anyEntriesFixable(preupgradeReports),
  'should detect a fixable entry in reports': () =>
    anyEntriesFixable(reportsWithRemediations),
  'should collect ids for job invocation from reports': () =>
    idsForInvocationFromReports(preupgradeReports),
  'should return entries page': () =>
    entriesPage(flattenEntries(preupgradeReports), { page: 2, perPage: 3 }),
  'should filter entries by title': () =>
    filterEntries('title', 'broken', flattenEntries(preupgradeReports)),
  'should filter entries by severity': () =>
    filterEntries('severity', 'medium', flattenEntries(preupgradeReports)),
  'should filter entries by host': () =>
    filterEntries('hostname', 'foo', flattenEntries(preupgradeReports)),
  'should filter entries by remediations': () =>
    filterEntries('fix', 'command', flattenEntries(reportsWithRemediations)),
  'should filter entries by inhibitor': () =>
    filterEntries('inhibitor', 'yes', flattenEntries(preupgradeReports)),
  'should return fixable entries': () =>
    fixableEntries(reportsWithRemediations),
  'should sort entries by title in ascending order': () =>
    sortEntries(flattenEntries(preupgradeReports), {
      attribute: 'title',
      order: 'asc',
    }),
  'should sort entries by title in descending order': () =>
    sortEntries(flattenEntries(preupgradeReports), {
      attribute: 'title',
      order: 'desc',
    }),
  'should sort entries by hostname': () =>
    sortEntries(flattenEntries(preupgradeReports), {
      attribute: 'hostname',
      order: 'asc',
    }),
  'should sort entries by severity': () =>
    sortEntries(flattenEntries(preupgradeReports), {
      attribute: 'severity',
      order: 'asc',
    }),
  'should sort entries by remediations': () =>
    sortEntries(flattenEntries(reportsWithRemediations), {
      attribute: 'fix',
      order: 'asc',
    }),
};

describe('PreupgradeReportsHelpers', () =>
  testSelectorsSnapshotWithFixtures(fixtures));
