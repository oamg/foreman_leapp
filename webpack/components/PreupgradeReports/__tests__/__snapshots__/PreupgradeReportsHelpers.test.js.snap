// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreupgradeReportsHelpers should collect ids for job invocation 1`] = `
Object {
  "entryIds": Array [],
  "hostIds": Array [],
}
`;

exports[`PreupgradeReportsHelpers should detect a fixable entry in reports 1`] = `true`;

exports[`PreupgradeReportsHelpers should detect empty object 1`] = `true`;

exports[`PreupgradeReportsHelpers should detect entry is fixable 1`] = `true`;

exports[`PreupgradeReportsHelpers should detect entry is not fixable when not having a command 1`] = `false`;

exports[`PreupgradeReportsHelpers should detect entry is not fixable when not having remediations 1`] = `false`;

exports[`PreupgradeReportsHelpers should detect no entry is fixable 1`] = `false`;

exports[`PreupgradeReportsHelpers should detect non-empty object 1`] = `false`;

exports[`PreupgradeReportsHelpers should filter entries by host 1`] = `
Array [
  Object {
    "hostname": "foo.example.com",
    "id": 44,
    "severity": "high",
    "title": "Octocat is not happy",
  },
  Object {
    "hostname": "foo.example.com",
    "id": 45,
    "severity": "low",
    "title": "Not enough credits",
  },
  Object {
    "hostname": "foo.example.com",
    "id": 46,
    "severity": "medium",
    "title": "SELinux is turned off",
  },
  Object {
    "hostname": "foo.example.com",
    "id": 47,
    "severity": "medium",
    "title": "Root password is too short",
  },
  Object {
    "hostname": "foo.example.com",
    "id": 49,
    "severity": "high",
    "title": "No chocolate chip cookies in cookie jar",
  },
]
`;

exports[`PreupgradeReportsHelpers should filter entries by remediations 1`] = `
Array [
  Object {
    "detail": Object {
      "remediations": Array [
        Object {
          "type": "command",
        },
      ],
    },
    "severity": "high",
    "title": "Octocat is not happy",
  },
]
`;

exports[`PreupgradeReportsHelpers should filter entries by severity 1`] = `
Array [
  Object {
    "hostname": "host.example.com",
    "id": 43,
    "severity": "medium",
    "title": "I am broken too",
  },
  Object {
    "hostname": "foo.example.com",
    "id": 46,
    "severity": "medium",
    "title": "SELinux is turned off",
  },
  Object {
    "hostname": "foo.example.com",
    "id": 47,
    "severity": "medium",
    "title": "Root password is too short",
  },
]
`;

exports[`PreupgradeReportsHelpers should filter entries by title 1`] = `
Array [
  Object {
    "hostname": "host.example.com",
    "id": 43,
    "severity": "medium",
    "title": "I am broken too",
  },
]
`;

exports[`PreupgradeReportsHelpers should return entries page 1`] = `
Array [
  Object {
    "hostname": "foo.example.com",
    "id": 45,
    "severity": "low",
    "title": "Not enough credits",
  },
  Object {
    "hostname": "foo.example.com",
    "id": 46,
    "severity": "medium",
    "title": "SELinux is turned off",
  },
  Object {
    "hostname": "foo.example.com",
    "id": 47,
    "severity": "medium",
    "title": "Root password is too short",
  },
]
`;

exports[`PreupgradeReportsHelpers should return flattened entrues 1`] = `
Array [
  Object {
    "hostname": "host.example.com",
    "id": 42,
    "severity": "Too severe to talk about",
    "title": "Fix me!",
  },
  Object {
    "hostname": "host.example.com",
    "id": 43,
    "severity": "medium",
    "title": "I am broken too",
  },
  Object {
    "hostname": "foo.example.com",
    "id": 44,
    "severity": "high",
    "title": "Octocat is not happy",
  },
  Object {
    "hostname": "foo.example.com",
    "id": 45,
    "severity": "low",
    "title": "Not enough credits",
  },
  Object {
    "hostname": "foo.example.com",
    "id": 46,
    "severity": "medium",
    "title": "SELinux is turned off",
  },
  Object {
    "hostname": "foo.example.com",
    "id": 47,
    "severity": "medium",
    "title": "Root password is too short",
  },
  Object {
    "hostname": "foo.example.com",
    "id": 49,
    "severity": "high",
    "title": "No chocolate chip cookies in cookie jar",
  },
]
`;
